internal/pkg/siderolink/trustd/server.go

```diff
@@ -89,11 +92,29 @@ func (h *handler) Certificate(ctx context.Context, in *securityapi.CertificateRe
 		return nil, status.Error(codes.PermissionDenied, "peer address is not TCP")
 	}
 
-	secretsBundle, err := getSecretsBundle(ctx, h.state, tcpAddr.IP.String())
+	clusterSecrets, err := getSecrets(ctx, h.state, tcpAddr.IP.String())
 	if err != nil {
 		return nil, err
 	}
 
+	secretsBundle, err := omni.ToSecretsBundle(clusterSecrets.TypedSpec().Value.GetData())
+	if err != nil {
+		return nil, err
+	}
+
+	acceptedCAs := []*x509.PEMEncodedCertificate{{Crt: secretsBundle.Certs.OS.Crt}}
+
+	if clusterSecrets.TypedSpec().Value.RotationPhase != specs.ClusterSecretsRotationStatusSpec_OK {
+		var rotateSecretsBundle *secrets.Bundle
+
+		rotateSecretsBundle, err = omni.ToSecretsBundle(clusterSecrets.TypedSpec().Value.GetRotateData())
+		if err != nil {
+			return nil, err
+		}
+
+		acceptedCAs = append(acceptedCAs, &x509.PEMEncodedCertificate{Crt: rotateSecretsBundle.Certs.OS.Crt})
+	}
```

Comment:
I'll start with easy ones. I would propose to build the list of accepted CAs in the `ClusterSecrets` controller.

```suggestion
	secretsBundle, err := getSecretsBundle(ctx, h.state, tcpAddr.IP.String())
	if err != nil {
		return err
	}
	
	acceptedCAs := clusterSecrets.TypedSpec().Value.AcceptedCAs
```

Reply:
We'd need acceptedCAs for each component. In this case this is CAs for Talos.

Reply:
I think we can also combine `acceptedCAs` into a single `[]byte` slice to avoid doing join in all the places.